portable/ARMv8M/non_secure/port.c

```diff
@@ -1259,3 +1388,61 @@ BaseType_t xPortIsInsideInterrupt( void )
     return xReturn;
 }
 /*-----------------------------------------------------------*/
+
+#if ( configASSERT_DEFINED == 1 && portARM_CORTEX_M23 != 1 )
+    void vPortValidateInterruptPriority( void )
+    {
+        uint32_t ulCurrentInterrupt;
+        uint8_t ucCurrentPriority;
+
+        /* Obtain the number of the currently executing interrupt. */
+        __asm volatile ( "mrs %0, ipsr" : "=r" ( ulCurrentInterrupt )::"memory" );
```

Comment:
There may also be an issue here for ISRs executing in the non-secure state.   It seems IPSR won't indicate which interrupt is active when read from non-secure side.